### ISSUE TITLE
Doc to install lua-http on macos and lua 5.3

### ISSUE THREAD

**Issue by octplane** [daurnimator/lua-http#142]
This is probably a known issue, but there is not clear doc on how to do that correctly:

With `lua 5.3` from brew, the install command that succeeds on my machine is:

```shell
 luarocks install http CRYPTO_DIR=/usr/local/opt/openssl/ CRYPTO_INCDIR=/usr/local/opt/openssl/include OPENSSL_DIR=/usr/local/opt/openssl/ OPENSSL_INCDIR=/usr/local/opt/openssl/include
```

Thanks for making lua-http!

**Comment by daurnimator** [comment]
The problem is that it changes somewhat regularly and not in a way that is easily coordinated with the lua-http release cycle:

  - it's not lua-http that needs openssl, but lua-http's dependencies
  - there are several different ways to install openssl, even on just OSX
  - until recently, apple shipped openssl themselves (but an old version)
  - brew is quite configurable; and they have moved the location before

So I'm not sure what I can do to avoid having potentially outdated instructions around.

**Comment by RyanSquared** [comment]
I have absolutely no clue what's up with the openssl situation on macOS to the point where I'm getting a new computer, part of the justification being that I just don't want to deal with it. Unfortunately, it's very much a "per-system" situation.

**Comment by octplane** [comment]
Yeah, I actually opened the bug so that people like me (and probably me later) remember this might not be as straightforward as `luarocks install http` on macOS :)

**Comment by daurnimator** [comment]
One solution (given that you seem to have decided on brew) would be to package lua-http in brew?

**Comment by latenitefilms** [comment]
Thanks heaps @octplane ! This issue was very helpful!

**Comment by javierfernandes** [comment]
Still does the trick ! Thanks 🙏  !

**Comment by daurnimator** [comment]
Is there anyone that would like to create/submit a brew package for lua-http?

**Comment by NikoManiac** [comment]
Update: if you install openssl now, Brew will install openssl@3 on your mac.  The LuaRocks will install failed by use openssl@3 and error is: `Could not find header file for CRYPTO`.

This is workaround on my machine:
`luarocks install http CRYPTO_DIR=/usr/local/opt/openssl@1.1/ CRYPTO_INCDIR=/usr/local/opt/openssl@1.1/include OPENSSL_DIR=/usr/local/opt/openssl@1.1/ OPENSSL_INCDIR=/usr/local/opt/openssl@1.1/include`

**Comment by vvvvv** [comment]
For anyone coming here in 2022, this is still an issue.  

This is a workaround on my machine macbook m1

`luarocks install http CRYPTO_DIR=/opt/homebrew/opt/openssl@1.1 CRYPTO_INCDIR=/opt/homebrew/opt/openssl@1.1/include OPENSSL_DIR=/opt/homebrew/opt/openssl@1.1 OPENSSL_INCDIR=/opt/homebrew/opt/openssl@1.1/include`

**Comment by daurnimator** [comment]
> ```shell
> openssl@1.1
> ```

openssl 1.1 is quite old at this point. consider using a newer release?

**Comment by alextes** [comment]
During installation one of the steps lists where it was looking for openssl stuff, and does not list the place where homebrew installs the shared openssl things.

```sh
Error: Failed installing dependency: https://luarocks.org/cqueues-20200726.54-0.src.rock - Could not find header file for CRYPTO
  No file openssl/crypto.h in /usr/local/include
  No file openssl/crypto.h in /usr/include
  No file openssl/crypto.h in /include
  No file openssl/crypto.h in /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
  No file openssl/crypto.h in /usr/local/opt/include
```

Is that a luarocks or a cqueues issue? One of them is clearly unaware of the location where all macOS users will have their include dir, which is a lot of people! I can open something in one of those repos if someone has the answer.

For this repo I think the natural thing to do is to either note something in the install section or maybe a troubleshooting that mentions install on macOS requires additional steps. Even if only linking this issue. Installing an http lib is after all fairly common 😅 , and this appears to be the most popular option. At the same time, I'd also say it's fair to do nothing until something is fixed upstream. The lib which depends on openssl should know how to find it, not this lib. Nevertheless, this repo will probably face the most stuck end-users.

**Comment by daurnimator** [comment]
> One of them is clearly unaware of the location where all macOS users will have their include dir, which is a lot of people!

It's a luarocks configuration issue: you'd need to tell luarocks about where `brew` is putting things.
As upstream luarocks doesn't really know anything about brew, it's more of a question for whereever you got your luarocks config from.... which is probably brew?
i.e. it's a missing feature of the brew packaging of luarocks; in that it doesn't tell it about how brew works.
